src/semidiscretization/semidiscretization_hyperbolic.jl

```diff
@@ -210,6 +214,96 @@ function digest_boundary_conditions(boundary_conditions::AbstractArray, mesh, so
     throw(ArgumentError("Please use a (named) tuple instead of an (abstract) array to supply multiple boundary conditions (to improve performance)."))
 end
 
+function check_periodicity_mesh_boundary_conditions(mesh::TreeMesh{1},
+                                                    boundary_conditions)
+    @unpack x_neg, x_pos = boundary_conditions
+    if isperiodic(mesh.tree, 1) && (x_neg != BoundaryConditionPeriodic() ||
+        x_pos != BoundaryConditionPeriodic())
+        @warn "For periodic mesh non-periodic boundary conditions are supplied.
+         The boundary conditions will be ignored and periodic boundary conditions are used."
+    end
+end
+
+function check_periodicity_mesh_boundary_conditions(mesh::StructuredMesh{1},
+                                                    boundary_conditions)
+    @unpack x_neg, x_pos = boundary_conditions
+    if isperiodic(mesh, 1) && (x_neg != BoundaryConditionPeriodic() ||
+        x_pos != BoundaryConditionPeriodic())
+        @warn "For periodic mesh non-periodic boundary conditions are supplied.
+         The boundary conditions will be ignored and periodic boundary conditions are used."
+    end
+end
```

Comment:
Isn't this the same as the version for the `TreeMesh{1}`? If so, can't we use the same method above specializing on `Union{TreeMesh{1}, StructuredMesh{1}}`?

Reply:
No, for `TreeMesh` types we call `isperiodic(mesh.tree)` vs `isperiodic(mesh)` for structured meshes.

Reply:
Maybe we need another isperiodic then?

Reply:
Yes, please.